gradle-jdks-certs/src/main/resources/gradle-jdk-resolver.sh

```diff
@@ -0,0 +1,128 @@
+#!/bin/sh
+#
+# (c) Copyright 2024 Palantir Technologies Inc. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+##############################################################################
+#
+#   Gradle jdk set up script for POSIX generated by gradle-jdks.
+#
+#   This script does the following:
+#   (1) Downloads the corresponding JDK distribution based on the majorVersion=`gradle/gradle-jdk-major-version.txt`
+#     and the distribution_url=`gradle/jdks/${majorVersion}/${os}/${arch}/download_url.txt`
+#   (2) Installs the distribution in `$HOME/.gradle/gradle-jdks/${local_path}` based on the
+#     local_path=`gradle/jdks/${majorVersion}/${os}/${arch}/local_path.txt`
+#   (3) Calls the java class `JdkSpecCertSetup` that will set up the certificates for the locally installed distribution
+#   (4) Sets up the JAVA_HOME env variable to the currently installed JDK
+#
+#
+#   Important for running:
+#   This script requires all of these POSIX shell features:
+#         * functions;
+#         * expansions «$var», «${var}», «${var%suffix}», and «$( cmd )»;
+#         * various built-in commands including «command» and «set».
+#
+##############################################################################
+
```

Comment:
Wonder if we should `set -eu` for a bit of extra safety? So it exits on commands that return non-zero exit codes (`-e`) rather than continues, also exits on unset variables (`-u`).

Also, probably don't want it in prod, but while devving you can use `set -x` to print out each line of the file before execution happens to trace what is happening.

Reply:
+1  I was trying the `set -eu` but I was getting a weird cryptic failure for a test so I dropped it. 

> set -x

nice! didn't know about this

Reply:
The errors might be busybox (in alpine) or some shell not supporting it. It's normally busybox - it's so stripped down it has basically not features.